File: leetcode/20211022/majorityElement.go
package _test

func majorityElement(nums []int) (ans []int) {
	cnt := map[int]int{}
	for _, v := range nums {
		cnt[v]++
	}

	max := len(nums) / 3

	for k, v := range cnt {
		if v > max {
			ans = append(ans, k)
		}
	}

	return
}

File: leetcode/20211022/majorityElement.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer, Integer> cnt = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            if (cnt.containsKey(nums[i])) {
                cnt.put(nums[i], cnt.get(nums[i]) + 1);
            } else {
                cnt.put(nums[i], 1);
            }
        }

        List<Integer> ans = new ArrayList<Integer>();

        for (int x : cnt.keySet()) {
            ans.add(cnt.get(x));
        }

        return ans;
    }
}